parser.y

```diff
@@ -1961,6 +2007,13 @@ ConstraintElem:
 			Refer:		$8.(*ast.ReferenceDef),
 		}
 	}
+|	"CHECK" '(' Expression ')' EnforcedOrNotOpt
+	{
+		/* Nothing to do now */
+		$$ = nil
```

Comment:
I think we should still parse this into a valid `ast.Constraint` instance (like the FOREIGN KEY above), so we don't need to add all those `nil` checking.

Reply:
if we return `*ast.Constraint{}` here, We need special handling of `Constraint.Restore`.
and I don't know what will tidb do when it finds an `ast.Constraint` with `type=0`

Reply:
I mean we do support CHECK in constraint in the AST and add an `ast.ConstraintCheck` type constant (like FOREIGN KEY and FULLTEXT), but we ignore or warn about this on TiDB level.

Reply:
fine

Reply:
PTAL again

Reply:
tidb will ignore `ConstraintCheck`,
see:https://github.com/pingcap/tidb/blob/source-code/ddl/ddl_api.go#L901

so, I think parser should warn this